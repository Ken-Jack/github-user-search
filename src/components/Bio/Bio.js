// @flow

import React from 'react';
import {connect} from 'react-fela';
import linkify from 'util/linkify';

type Props = {
  text: string | null,
  styles: Object,
};

function Bio({text, styles}: Props) {
  if (!text) {return null;}
  return (
    <p
      className={styles.Bio_root}
      dangerouslySetInnerHTML={{__html: linkify(text)}}
    />
  );
}

const styles = {
  Bio_root: () => ({
    fontSize: 16,
    lineHeight: 1.3,
  }),
};

export default connect(styles)(Bio);
