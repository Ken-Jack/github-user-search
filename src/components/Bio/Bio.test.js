import React from 'react';
import felaSnapshot from 'test-util/fela-snapshot';
import Bio from './Bio';

describe('Component: Bio', () => {

  it('should render', () => {
    const component = felaSnapshot(
      <Bio
        text="This is some bio text with @link"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should not render if props is absent', () => {
    const component = felaSnapshot(
      <Bio />
    );
    expect(component).toMatchSnapshot();
  });

});
