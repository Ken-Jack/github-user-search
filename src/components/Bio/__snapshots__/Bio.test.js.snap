// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Bio should not render if props is absent 1`] = `
Object {
  "component": null,
  "styles": "
    .a{font-size:16}
    .b{line-height:1.3}",
}
`;

exports[`Component: Bio should render 1`] = `
Object {
  "component": <p
    className="a b"
    dangerouslySetInnerHTML={
      Object {
        "__html": "This is some bio text with <a href=\\"http://github.com/link\\">@link</a>",
      }
    }
  />,
  "styles": "
    .a{font-size:16}
    .b{line-height:1.3}",
}
`;
